File: Sunrin-ICPC-2022/Final-A/FA.py
n, m = map(int, input().split())
cnt = [0] * 3
for i in range(n): input()
for i in range(m):
    c = input().split()[2]
    if c == 'R': cnt[0] += 1
    elif c == 'B': cnt[1] += 1
    else: cnt[2] += 1

i = 0
while True:
    if cnt[2] > 0: cnt[2] -= 1
    elif cnt[i%2] > 0: cnt[i%2] -= 1
    else:
        if i % 2 == 0: print('jhnan917')
        else: print('jhnah917')
        break
    i += 1

File: Sunrin-ICPC-2022/Final-A/FA.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

let line = 0;
const temp = input[line++].split(' ');
const n = +temp[0], m = +temp[1];
let cnt = [0, 0, 0];
line += n;
for(let i=0; i<m; i++){
    const c = input[line++].split(' ')[2];
    if(c == 'R') cnt[0]++;
    else if(c == 'B') cnt[1]++;
    else cnt[2]++;
}
for(let i=0; ; i++){
    if(cnt[2] > 0) cnt[2]--;
    else if(cnt[i%2] > 0) cnt[i%2]--;
    else{
        if(i % 2 == 0) console.log('jhnan917');
        else console.log('jhnah917');
        break;
    }
}